pyartifactory/objects.py

```diff
@@ -231,6 +169,91 @@ def unlock(self, name: str) -> None:
         logger.debug("User % successfully unlocked", name)
 
 
+class ArtifactoryPermission(ArtifactoryObject):
+    """Models an artifactory permission."""
+
+    def __init__(self, artifactory: AuthModel) -> None:
+        super().__init__(artifactory)
+        if self._api_version == 2:
+            self._uri = "v2/security/permissions"
+        if self._api_version == 1:
+            self._uri = "security/permissions"
+
+    def create(
```

Comment:
Any reason to have both create and update? Should they be included into one, with upsert-argument instead?

Reply:
In this library, we follow the Artifactory API endpoints that are `CREATE, GET, UPDATE and DELETE` options. We do not want to disturb the users by including create and update into one call.
https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API+V2#ArtifactoryRESTAPIV2-CreatePermissionTarget